src/pasta/fields/fp.rs

```diff
@@ -164,69 +164,61 @@ impl_binops_additive!(Fp, Fp);
 impl_binops_multiplicative!(Fp, Fp);
 
 /// INV = -(p^{-1} mod 2^64) mod 2^64
-const INV: u64 = 0xa14064e1ffffffff;
+const INV: u64 = 0x992d30ecffffffff;
 
 /// R = 2^256 mod p
 const R: Fp = Fp([
-    0x1c3ed159fffffffd,
-    0xf5601c89bb41f2d3,
+    0x34786d38fffffffd,
+    0x992c350be41914ad,
     0xffffffffffffffff,
     0x3fffffffffffffff,
 ]);
 
 /// R^2 = 2^512 mod p
 const R2: Fp = Fp([
-    0x280c9c4000000010,
-    0x91a4409b5400af74,
-    0xdd7b28e19094c659,
-    0xc8ad9107ccca0e,
+    0x8c78ecb30000000f,
+    0xd7d30dbd8b0de0e7,
+    0x7797a99bc3c95d18,
+    0x96d41af7b9cb714,
 ]);
 
 /// R^3 = 2^768 mod p
 const R3: Fp = Fp([
-    0x98fb3d144380a737,
-    0xf9fdbeb55b7eb87c,
-    0x63f75cb999eafa89,
-    0x217cb214ebb8fc72,
+    0xf185a5993a9e10f9,
+    0xf6a68f3b6ac5b1d1,
+    0xdf8d1014353fd42c,
+    0x2ae309222d2d9910,
 ]);
 
-/// `GENERATOR = 5 mod p` is a generator of the `p - 1` order multiplicative subgroup, and
-/// is also a quadratic non-residue.
+/// `GENERATOR = 5 mod p` is a generator of the `p - 1` order multiplicative
+/// subgroup, or in other words a primitive root of the field.
 const GENERATOR: Fp = Fp::from_raw([
     0x0000_0000_0000_0005,
     0x0000_0000_0000_0000,
     0x0000_0000_0000_0000,
     0x0000_0000_0000_0000,
 ]);
 
-const S: u32 = 33;
+const S: u32 = 32;
 
 /// GENERATOR^t where t * 2^s + 1 = p
 /// with t odd. In other words, this
 /// is a 2^s root of unity.
-///
-/// `GENERATOR = 5 mod p` is a generator
-/// of the p - 1 order multiplicative
-/// subgroup.
 const ROOT_OF_UNITY: Fp = Fp::from_raw([
-    0x53de9f31b88837ce,
-    0xff46e8f3f3ea99d6,
-    0xf624f2eaaf8c2d57,
-    0x2ae45117890ee2fc,
+    0xbdad6fabd87ea32f,
+    0xea322bf2b7bb7584,
+    0x362120830561f81a,
+    0x2bce74deac30ebda,
```

Comment:
I did not check this constant, but I'm satisfied by its unit test.

Reply:
Checked all the constants up to and including this one.